ISPN-15124 Better describe SQL store limitations (#11234)
infinispan/infinispan: documentation/src/main/asciidoc/topics/con_cache_store_sql.adoc

For composite keys and values, you must provide {brandname} with Protobuf schema (`.proto` files) that describe the keys and values.
With {brandname} Server you can add schema through the {brandname} Console or Command Line Interface (CLI) with the [command]`schema` command.

[WARNING]
====
Since a SQL cache store is designed to be used with an existing database table it would not store any metadata, which includes expiration, segments and versioning metadata.

> Optimistic transactional cache and asynchronous cross-site replication won't work with the SQL store due to the lost of versions.

Reviewer: `s/lost/loss`

> Optimistic transactional cache and asynchronous cross-site replication won't work with the SQL store due to the lost of versions.

Reviewer 2: ```suggestion
Optimistic transactional cache and asynchronous cross-site replication won't work with the SQL store due to not storing versions.
```
Just to make it a bit more explicit.
